src/client/app/containers/BarChartContainer.ts

```diff
@@ -40,44 +41,60 @@ function mapStateToProps(state: State) {
 		const byMeterID = state.readings.bar.byMeterID[meterID];
 		if (byMeterID !== undefined && byMeterID[timeInterval.toString()] !== undefined &&
 			byMeterID[timeInterval.toString()][barDuration.toISOString()] !== undefined) {
-			const readingsData = byMeterID[timeInterval.toString()][barDuration.toISOString()][unitID];
-			if (readingsData !== undefined && !readingsData.isFetching) {
-				const label = state.meters.byMeterID[meterID].identifier;
-				const colorID = meterID;
-				if (readingsData.readings === undefined) {
-					throw new Error('Unacceptable condition: readingsData.readings is undefined.');
+			let meterArea = state.meters.byMeterID[meterID].area;
+			// we either don't care about area, or we do in which case there needs to be a nonzero area
+			if (!state.graph.areaNormalization || (meterArea > 0 && state.meters.byMeterID[meterID].areaUnit != AreaUnitType.none)) {
```

Comment:
The reason the statements are done like this is because if area normalization is enabled, then meters that do not satisfy the condition should not be graphed at all, and will fail this check.